### ISSUE TITLE
While installing solder 'out of memory' error 1&1 webserver

### ISSUE THREAD

**Issue by mymindstorm** [TechnicPack/TechnicSolder#290]
While installing solder I got this:
```php
X-Powered-By: PHP/5.5.20
Content-type: text/html

Warning: Composer should be invoked via the CLI version of PHP, not the cgi-fcgi SAPI
Loading composer repositories with package information
Installing dependencies (including require-dev)
<br />
<b>Fatal error</b>:  Out of memory (allocated 77332480) (tried to allocate 1091657 bytes) in <b>phar:///homepages/34/d523839286/htdocs/modpack/composer.phar/src/Composer/Repository/ComposerRepository.php</b> on line <b>294</b><br />
```
I used:
```
php5.5 composer.phar install --no-interaction
```
I don't know why this is happening, because I have 100 GB available. I'm running a Linux server with 1&1, php 5.5 is set as default but when I use `php` it uses php 4.4.9.

**Comment by sct** [comment]
You have 100GB of ram? I doubt it. You are running out of ram while composer is trying to download all of the dependencies. Increase the amount of memory PHP can allocate in PHP.ini

**Comment by mymindstorm** [comment]
No I thought this was a space error, I have 100 GB of space. 1&1 doesn't allow me to allocate any more memory
![image](https://cloud.githubusercontent.com/assets/6856311/5993263/095d3e4c-aa10-11e4-8871-f645bd2465e8.png)

is there a workaround?

**Comment by mymindstorm** [comment]
I meant space, I thought it ran out of space. I'm new to PHP so I'll see if that works.

**Comment by GenPage** [comment]
Make sure you have `php-cli` or `php5-cli` installed

**Comment by sct** [comment]
Hes probably forcing it to run through a cron or something.

**Comment by GenPage** [comment]
I just re-read his comment, and it says 1&1 does not allowing increasing the PHP memory limit over 30MB. That will do it.

**Comment by mymindstorm** [comment]
@GenPage I have it as `php5.5-cli`
@sct No i'm not even trying a cron

**Comment by mymindstorm** [comment]
@GenPage It says that only for the older packages, I have a newer package. My current limit is 61M if you looked at the config

**Comment by GenPage** [comment]
Still even 61 MB isn't a whole lot. I suggest running composer in a local setup first and then uploading the parent folder

**Comment by mymindstorm** [comment]
@GenPage ok I'll try that

**Comment by mymindstorm** [comment]
Now I get this
`Fatal error: require(): Failed opening required '/homepages/34/d523839286/htdocs/modpack/bootstrap/../vendor/autoload.php' (include_path='.:/usr/lib/php5.5') in /homepages/34/d523839286/htdocs/modpack/bootstrap/autoload.php on line 17`
when I go to my website. I just finished running the script, am I missing any files?
![image](https://cloud.githubusercontent.com/assets/6856311/5993932/d1c00b88-aa26-11e4-8050-d657e8fd0951.png)

**Comment by spannerman79** [comment]
@GenPage when you state running composer on local setup and upload parent folder you mean after going through running;
    git clone https://github.com/TechnicPack/TechnicSolder.git
    php composer.phar install --no-dev --no-interaction
etc
Then upload the TechnicSolder/ directory?

**Comment by mymindstorm** [comment]
@spannerman79 yes that's what he meant; gave me internal server error

**Comment by spannerman79** [comment]
Ok just so that people can know - and that the solder.io documentaiton can be updated;
I tried the above out via Dreamhost and it does not work while using shared hosting.

The reason why i tried Dreamhost as they are the only real shared web hosting that allows for a sort of shell access to your account.

The support people over at Dreamhost even lifted the temp memory limit off just so that i can compile and perform the pho composer part.  The moment I did that i sent a message to them and they reimposed the PHP memory limit and Solder wouldn't not function.

Once again to make it clear - Using shared webhosting does NOT work.
This is not a complaint but something to enforce in documentation on GitHub (in readme.md) and on solder.io
